lib/pbench/agent/tool_meister.py

```diff
@@ -1311,125 +1355,83 @@ def _send_directory(self, directory, uri, ctx):
         target_dir = directory.name
         parent_dir = directory.parent
         tar_file = parent_dir / f"{target_dir}.tar.xz"
-        o_file = parent_dir / f"{target_dir}.tar.out"
-        e_file = parent_dir / f"{target_dir}.tar.err"
-        try:
-            # Invoke tar directly for efficiency.
-            with o_file.open("w") as ofp, e_file.open("w") as efp:
-                cp = subprocess.run(
-                    [
-                        self.tar_path,
-                        "--create",
-                        "--xz",
-                        "--force-local",
-                        f"--file={tar_file}",
-                        target_dir,
-                    ],
-                    cwd=parent_dir,
-                    stdin=None,
-                    stdout=ofp,
-                    stderr=efp,
+
+        tar_process = self._create_tar(target_dir, tar_file)
+        if tar_process.returncode != 0:
+            # Re-try creating tar ball by suppressing all the warnings
+            tar_process = self._create_tar(target_dir, tar_file, ignore_warning=True)
+            if not tar_process:
+                # Tar ball creation failed even after suppressing all the warnings,
+                # we will now proceed to send an empty tar ball
+                self._create_tar(
+                    Path("/dev/null"), tar_file, cwd=parent_dir, ignore_warning=True
                 )
```

Comment:
Are you sure you want to go to the trouble of trying to create an empty archive, rather than just creating a completely empty file?  (I mean, at this point, the `tar` command has failed twice already...it might fail a third time....)

If this third call fails, will we hang?

Reply:
> If this third call fails, will we hang?

Yes, if this fails then the TDS hangs 

> Are you sure you want to go to the trouble of trying to create an empty archive, rather than just creating a completely empty file? (I mean, at this point, the `tar` command has failed twice already...it might fail a third time....)

The `put_document` of the tool data sink write this file as `tar.xz` on the controller, can we just send a regular file over to the tds and will the tds write it as a tar file or will that produce an exception? 🤔

Reply:
```
[root@dbutenho-dev ~]# touch foo.tar.xz
[root@dbutenho-dev ~]# tar xvf foo.tar.xz 
tar: This does not look like a tar archive
xz: (stdin): File format not recognized
tar: Child returned status 1
tar: Error is not recoverable: exiting now
[root@dbutenho-dev ~]# echo $?
2
```

So, yeah; the `PUT` will run into a problem unpacking the file, causing `PUT` to terminate with a 500 status ("Failed to extract tar ball"), which will cause `_send_directory` to log a message and count a failure. Looks like the `PUT` will *not* update `_tm_tracking[hostname]` from "waiting" to "dormant", so the TDS will continue waiting for an artifact, and we haven't affected the hang.

At least passing the empty (real) tarball means that TDS can unpack and track receipt. Arguably, when TDS gets an upload that it can't unpack it should mark that host "failed" in the tracker and we should stop waiting. But, again, it'd be better to be able to skip the `PUT` entirely in this case and simply Redis a failure (yes, I've verbified that noun) that the TDS wait cycle will recognize as "no soup for you" and move on.

Reply:
> Arguably, when TDS gets an upload that it can't unpack it should mark that host "failed" in the tracker and we should stop waiting

Yeah ideally we would want to do that, but its bit tricky to do it outside the tds because there is a watcher thread in tds that change the tracking status to dormant after the put finishes. I will put a TODO comment in our code to address this thing separately. This will probably involve changes to how TM and TDS communicate with each other.

Reply:
If it sounded like I was trying to push you into fixing that now, I apologize; what I was intending was more to justify your temporary workaround... with acknowledgement that there's a better way. Peter already entered #2709 to approach this problem.